Introduce internationalization (i18n) policy (#51292)
mistercrunch/Airflow: airflow-core/src/airflow/ui/src/i18n/README.md

Reviewer: Do we want to add internal hyperlinks to sections in the file?

Author: Shouldn't be harmful - it's common in other MD files.

 under the License.
 -->

> # i18n Policy

Reviewer 2: ```suggestion
# Internationalization (i18n) Policy
```

not sure whether it's more clear this way 🤔

Reviewer 3: Agreed, we can do this

## 1. Purpose & scope

This document outlines the policy for internationalization (i18n) in Apache Airflow, detailing the lifecycle of translations within the project.
This policy aims to avoid inconsistencies, maintenance issues, unclear ownership, and to ensure translation quality.
The scope of this policy is applied to:

- Each supported language included in the `airflow-core/src/airflow/ui/src/i18n/locales` directory of the Apache Airflow project, or any suggested new translation.
- Individuals responsible for maintaining these translations by any of the roles defined below.
- Contributors who apply changes in the default language (English) that may affect translations.

## 2. Definitions

> **i18n** - Internationalization, the process of designing a software application so that it can be adapted to various languages and regions without engineering changes.

Reviewer 4: Does it make sense to point to the article in Wikipedia with definitions?
```suggestion
**i18n** - Internationalization, the process of designing a software application so that it can be adapted to various languages and regions without engineering changes. See also ([Wikipedia: Internationalization and localization](https://en.wikipedia.org/wiki/Internationalization_and_localization))
```

> **Supported translation** - A translation that has been officially accepted into the project, located in `airflow-core/src/airflow/ui/src/i18n/locales`.

Reviewer 4: Questioning myself: We make the rules here for the core UI... if we later start translating FAB UI or SimpleAuthManager... then we ajust the rule for these components managed in monorepo (potentially in future)?

(I would propose to start here and extend the definition if other compoentns e.g. airflowctl / FAB are translated if the rule is replicated... not to over-complicate here?

Reviewer 3: Agreed, start small, and build over it!

Author: I'll add a note about that in the scope

**Default language** - The language used by default, and as a fallback to all other languages (English).

**Translation owner** - A designated individual responsible for the maintenance and quality for a supported translation.

> **Code owner** - An Apache Airflow committer designated in the `.github/CODEOWNERS` file for a supported translation.

Reviewer 4: ```suggestion
**Code owner** - An Apache Airflow committer designated in the `.github/CODEOWNERS` file for a supported translation. Only Code owners have write permissions to the repository and can be listed in `.github/CODEOWNERS`.
```

> **Code owner** - An Apache Airflow committer designated in the `.github/CODEOWNERS` file for a supported translation.

Reviewer 3: These will be same in most cases?

Author: Probably, but not necessarily - I made this distinction to allow non-committers to participate :)

**Translation sponsor** - An Apache Airflow committer who supports a non-comitter translation owner (e.g., by merging Pull Requests on their behalf).

> **Engaged translator** - An Apache Airflow contributor who actively participates in the translation process, yet is not a translation owner.

Reviewer 3: Do we need this additional terminology

Author: Nice to have IMO, and it's already being used in the CODEOWNERS file:
https://github.com/apache/airflow/blob/158dcb0169199a0c6758193b307c35bb97adab7a/.github/CODEOWNERS#L41

The idea is to encourage people to start contributing translations without having an "official" role,
so in future decisions they might be provided with it :)

## 3. Wording/Phrasing

- Unless specified explicitly, references to directories and files in the document refer to files in the `main` branch.
- Where emphasised by capital letters, The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in the RFC 2119.

## 4. Requirements

### 4.1. Translation ownership and code ownership

Each supported translation, except for the default language, MUST meet one of the following conditions at any given time (except for transition times in section 6.4.):

- Have at least one translation owner who is also a code owner.

> - Have at least one translation owner, with a translation sponsor assigned as a code owner.

Reviewer 4: I'd propose to request more than one just as a fallback and vacation delegate
```suggestion
- Have at least one translation owner, with a translation sponsor assigned as a code owner.

It is welcomed and desired to have more than one individual translation owner to enable peer reviews and cover off-periods of individuals.
```

### 4.2. Approval of translation owners candidates

> - Translation owners candidates MUST declare and demonstrate a sufficient level of proficiency in the target language for translation purposes, including technical terminology.

Reviewer 3: demonstrate how? :D

- Translation owners candidates, who are non-committers, MUST also meet the following criteria:
     - They are active long-term contributors to the Apache Airflow project at the time of request.
     - They have basic skills of working with GIT and GitHub, as well as modifying JSON translation files within their target language.
     - They have the support of a Apache Airflow committer who will act as a translation sponsor.
- Translation owners candidates MUST go through the approval process detailed in section 6.1.

### 4.3. Approval of new translations

To accept a new translation to the codebase, it MUST be approved through a review process discussed in section 6.2.

### 4.4. Resolution of translation conflicts

Translation conflicts MUST be resolved according to the procedures outlined in section 6.3.

### 4.5. Adding / rephrasing terms

> - When new terms are added to the default language, all translation owners SHOULD create a follow-up PR to comply with the changes in their assigned language within a reasonable time.

Reviewer 4: ```suggestion
- When new terms are added to the default language, all translation owners SHOULD create a follow-up PR to comply with the changes in their assigned language within a reasonable time.
    - In busy times with many parallel UI changes it is acceptable to batch changes together, latest prior a release the differences must be cleared.
```

Reviewer 5: I think we should also somehow involve release management here:

* what to do if we miss translations prior to release?
* who and when checks that? (we already have tool for that - see tooling section - it just needs to be run at some point in time.
* should release manager hold release before translations are complete or do we allow incomplete translations ? 

I have no single answers or proposals to those but I guess we should at least think what to do and involve relase managers @kaxil and @ephraimbuddy (and others who might step in) so that they are not surprised they need to do something or that they do not even know they should do something.

Reviewer: How are translators notified about new strings being added? For example on Crowdin, whenever source strings are modified one can get an email. Is there going to be a similar notification mechanism here?

Author: @potiuk I added a section about that in the version release docs, tl;dr:
1. We should do our best not to miss, but If we do - there's always the fall back to English.
2. Release managers should run the tool before releasing RC, and then sending a reminder urging code owners to ensure completion of translations before a due date.
3. IMO we should not hold, and allow incomplete translations up to X consecutive releases (I set `X=2`, but we could change this number if it's too strict). If the incomplete translation proceeds to X+1, it's an indication that the translation owners are inactive - and could be a valid reason for removing this translation like any other unused Airflow feature.

Reviewer 6: Suggestions look good. Falling back to English is a good option, but it could end up mixed support for a couple of releases if the language is inactive. Setting some boundaries is also a good option. 

I think making this flexibility could already take another load from release managers. We should automate the exclusion in a way that releases managers shouldn't handle anything manual. We can easily understand if the language is active and supported or inactive with simple tricks. This should also be a newsfragment to include, which won't be supported in the next Nth release.

Reviewer: Re batching translation changes. I don't know what the current plan is but how about this:
- There will be a separate branch for i18n. Translation PRs will always target this branch. The branch may have unique CI configurations or approval requirements (as detailed in this document).
- To avoid adding to PR pressure, especially around releases, the i18n branch will be merged into main right before a release is cut. If permitted by Apache rules, i18n->main merges will occur at some fixed interval such as weekly or nightly.
- If the English source changes, this will be synced into the i18n branch via a PR that potentially includes notifying all translators. A small added benefit is it will make it easy for translators to see when their involvement might be needed, by inspecting the commit graph.

Author: @Dev-iL interesting idea, but let's maybe roll for 1-2 releases and then discuss if it's really needed - the effort for maintaining another branch (inc. related CI) at this point seems quite high IMO.
We could also think of a version management for langauge packages, but this requires a careful design as well.

> [!NOTE]
> Tooling for detecting missing translations is available (see Tools & Resources section below).

### 4.6. Deprecating / refactoring terms

When existing terms are deprecated or refactored (key renamed/relocated but value unchanged), **the contributor initiating the change is responsible for updating all language files, and not the translation/code owner**. Automation through Breeze tooling should be used when available.

### 4.7. Approval of translation-related Pull Requests (PRs)

- If the code owner is also a translation owner of the respective translation:
    - Others' PRs should be approved and merged normally by them.
    - Their own PRs should be approved and merged normally by another committer, preferably another code owner in their locale (if such exists).
- Otherwise, if the code owner is sponsored:
    - They should merge the translation-related PRs only after they are approved by the translation owner.
- Other committers may review and approve translation-related PRs in any aspects, but they SHOULD NOT merge them without the approval of the translation owner and the consent of the code owner.

> ## 5. Roles & responsibilities

Reviewer 2: Should we move this section above? I was confused when reading 4 without these context

### 5.1. Translation owner

- Translation owners are responsible for the following, in their assigned supported translation, according to established quality standards and procedures stated below:
  - Ensuring translation remains up-to-date with source code changes in the default language.
  - Reviewing the language aspects of translation-related Pull Requests (PRs).
  - Resolving translation-related conflicts in PRs.
  - Ensuring translation reflects current language usage and terminology.

>   - Resolving translation-related issues and discussions.

Reviewer 4: Repeat the policies here in regards of neutral tone?

Author: Here I actually meant GitHub issues :)
Anyway, the neutral tone is implied from the quality standards mentioned in the beginning

  - Resolving translation-related conflicts in PRs, when needed.
  - Managing translation-related issues and discussions, when needed (e.g., closing issues).
- Translations sponsors who function as code owners, are also responsible for ensuring that the translation owner is active and able to maintain the translation. If not, they should act according to section 6.4.

### 5.3. Engaged translator

- Engaged translators do not have any formal responsibilities, but they are encouraged to contribute to translations by:
  - Suggesting improvements to existing translations.
  - Reporting issues or inconsistencies in translations.
  - Participating in discussions related to translations.
  - Assisting translation owners with their tasks, when needed.
  - Being 3rd party reviewers for translation-related PRs and conflicts, when needed.
- They may be mentioned in a comment in the `.github/CODEOWNERS` file.
- Suitable candidates for translation ownership may be suggested from engaged translators, upon their consent and approval by the procedure in section 6.1.

## 6. Procedures

### 6.1. Approval process of a translation owner

> - The designated code owner should post a thread in the dev. list to request for approval of the translation owner(s) for a supported translation.:

Reviewer 2: should we add a link to `dev. list`?

Reviewer 3: +1

> - The designated code owner should post a thread in the dev. list to request for approval of the translation owner(s) for a supported translation.:

Reviewer: ```suggestion
- The designated code owner should post a thread in the dev. list to request for approval of the translation owner(s) for a supported translation:
```

    - Approvals of a translation owner who is also the code owner can be done by a lazy consensus.
    - Approvals of translation owners who are non-committers require at least one binding vote of at least 1 PMC member, and no objections from committers/PMC.
- Within the thread, the code owner should demonstrate that the translation owner is suitable for the role, according to the requirements in section 4.2.

### 6.2. Approval process of a new translation

The following steps outline the process for approving a new translation:
> [!WARNING]

> > Before merging the PR, the committer should validate that files are in sync with the default language (English).

Reviewer 3: Could we have a precommit that does it?

Author: We have Jarek's script, I'll add a reference for that

- Creating a PR to add a new translation to the codebase ([see example](https://github.com/apache/airflow/pull/51258/files)), according to the standard and guidelines, which includes:

>     - The translation files in the `airflow-core/src/airflow/ui/src/i18n/locales/<LOCALE_CODE>` directory, where `<LOCALE_CODE>` is the code of the language (e.g., `fr` for French).

Reviewer: Might be worth mentioning the acceptable standard for these entries (e.g. ISO 639-1 or 639-3). Perhaps it's worth transitioning the existing languages to the most recent standard, `639-3`, before it becomes too much work).

>     - Making the required modifications in `airflow-core/src/airflow/ui/src/i18n/config.ts` ([see example](https://github.com/apache/airflow/pull/51258/files#diff-bfb4d5fafd26d206fb4a545a41ba303f33d15a479d21e0a726fd743bdf9717ff)) (should be done by the code owner).

Reviewer 4: I think can be done by anybody but need to be approved by a codeowner :-D
```suggestion
    - Making the required modifications in `airflow-core/src/airflow/ui/src/i18n/config.ts` ([see example](https://github.com/apache/airflow/pull/51258/files#diff-bfb4d5fafd26d206fb4a545a41ba303f33d15a479d21e0a726fd743bdf9717ff)).
```

    - Updating the `.github/CODEOWNERS` file to include the code owner(s).
      > [!NOTE]
      > When assigning a translation sponsor, there should be a comment in the `.github/CODEOWNERS` indicating who the translation owner(s) and the translation sponsor are.
- Apply procedure 6.1. to approve the translation owner, while announcing the introduction of the new language (including a link to the PR).
- Only after the steps above are completed, the PR for the new translation could be merged (by the requirements in section 4.7).

### 6.3. Translation conflict resolution

When a conflict arises in a translation-related PR, the following steps will be taken in order:

- The involved parties should first try to reach a consensus through discussion in the PR.
- If no consensus is reached, a translation owner may decide the outcome.
- If multiple translation owners are involved and cannot reach consensus, the code owner will decide. If the code owner is sponsored,
they should base their decision on a neutral source (e.g., a third-party opinion, translation tool, or LLM such as ChatGPT or Claude).

> - If the conflict is between code owners, a PMC member will be involved to resolve the conflict.

Reviewer 4: Does this not rather need to follow the VOTING process like in debates about other technical / code questions?

Author: This is the most fundamental issue in the base of this policy - how do you even run such a vote, when most of the audience is not proficient in the target language? :)

Reviewer 4: I hope there are no real "conflicts". Maybe just "opinons". But languages are usually not so strickt and multiple options are typically a matter of taste.
Anyway I assume PMC role is not needed but it should be rather a binding vote from all committers. Yeah and also if not proficient in the language. Will be the same but less people as PMC as well. Otherwise you could only roll a dice or create a fork of Airflow :-D

Author: I'm convinced, changed it to a decision by voting

Reviewer 7: I am not sure if we should start mailing lists votes on language specific terminology. I don't think this is what the mailing list is for. A simple jusgment call from any PMC member should be sufficient on the PR itself (like any code change we have).

Similarly, I don't think any code flavor "conflict" between 2 developer over a specific provider would have reach to a mailing list discussion. So in my prespective it's kinda the same

Author: Now, we need to solve a conflict by two PMC members 😅
I'm ok with both decisions - I don't expect that it would happen *too* often, but voting for a language aspect matter in th dev list feels indeed a bit odd.
Jens - what do you think?

Reviewer 4: I do not have a strong opinion on this. Other is also okay. Just wanted to make it as much similar like other conflict resolutions in Apache guidelines. But I also assume conflicts are hopefully very rare and for example if @TJaniF and me can not agree on terms, then she just need to fork and create "Schwizerdeutsch" (de-CH) :-)

TLDR: I am also okay with @eladkal 's opinion.

Author: Restored the original phrasing, I'll notify about that in the vote

### 6.4. Relinquishing translation/code ownership

> - When a translation owner asks to relinquish their role, or becomes inactive, and there are no other translation owners available, the code owner should follow these steps:

Reviewer 4: What is a codeowner just goes passive and does not fulfill the role? Would it be acceptable that a different code owner also starts the process asking about status and maintenance?

Reviewer: Perhaps "inactive" should come with an exact definition, i.e. along the lines of "at least 3 months of the target language not being updated after changes to source strings were introduced".

Author: I'll add a definition for inactivity (tl;dr - two consecutive releases with incomplete translations, or inactivity in Airflow for more than 12 months)

  - They should raise a PR for removal of the translation owner from the `.github/CODEOWNERS` file.
  - They should post a thread in the dev. list that the translation owner relinquishes their role, and that they are looking for a replacement.
  - If a replacement is found within a reasonable time, they should be approved according to the relevant steps in section 6.1 - and afterward the code owner should update the `.github/CODEOWNERS` file to reflect the new translation owner.

>   - Otherwise, the code owner should raise a vote in the dev. list for the removal if the translation from the codebase.

Reviewer: ```suggestion
  - Otherwise, the code owner should raise a vote in the dev. list for the removal of the translation from the codebase.
```

- When a translation sponsor asks to relinquish their role, and there is at least one translation owner - they should ask in the dev. list for assigning another sponsor instead.

## 7. Standards & guidelines

- Translations should be based on the default language (English). When translating a language that has already a similar translation supported
(e.g., Portuguese vs. Spanish), the other language might be used as a reference, but still the default language (English) should be the primary source for translations.
- Translations should be accurate, maintaining original meaning and intent.
- Translations should be complete, covering all terms and phrases in the default language.
- Translation of technical terminology should be consistent (for example: Dag, Task, Operator, etc.).
- Language should be polite and neutral in tone.
- Usage language that defies Apache Airflow's [code of conduct](http://airflow.apache.org/code-of-conduct/) is prohibited.

> [!NOTE]
> In case that local conventions requires deviation from the following guidelines, exceptions may be requested via PR or the dev list.

> 

Reviewer 2: ```suggestion
```

- Local conventions should be considered (e.g., date formats, number formatting, formal vs. informal tone, etc.)
- Formatting, placeholders, and variable substitutions must be preserved.

> [!WARNING]

> > In languages with different word order than English, or in Right-To-Left (RTL) languages, it is important to validate that the changes are properly reflected in the UI.

Reviewer: 1. What to do if they're not? Add/remove abbreviations? Take artistic freedom?
2. Should we require screenshots in PRs?
3. "languages with different word order than English" - there are too many languages that fit this description. I'm particularly worried about strings that mix multiple languages (and directions), e.g."ה-DAGים", which it is extremely easy to get wrong.

Author: 1. GitHub issue / PR for fixing it, like we do here the best (I'll add it) :)
2. I don't think that we should require that - but reviewers might definitely ask contributors if they checked that the PR is according to the quality standards (we could even later create a PR template for translations).
3. That's correct, so both contributors and reviewers should be cautious about it. Moreover, I think that even in latin-scripted language we could have UI issues for templated strings.

**Translation sponsor** - An Apache Airflow committer who supports a non-comitter translation owner (e.g., by merging Pull Requests on their behalf).

**Engaged translator** - An Apache Airflow contributor who actively participates in the translation process, yet is not a translation owner.

**Release manager** - according to the definition in the [Release Management Policy](../../../../../../dev/README_RELEASE_PROVIDERS.md#what-the-provider-distributions-are).

**Dev. list** - The Apache Airflow development mailing list: dev@airflow.apache.org.

**Inactive translation/code owner** — A translation owner/code owner is considered inactive if they meet either of the following criteria:

- The translation under their responsibility has remained incomplete for at least two consecutive releases.
- They have not participated in the Apache Airflow project for more than 12 months.

## 3. Wording/Phrasing

- Unless specified explicitly, references to directories and files in the document refer to files in the `main` branch.

> - Where emphasised by capital letters, The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",

Reviewer: ```suggestion
- Where emphasised by capital letters, the keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
```

> "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in the RFC 2119.

Reviewer: ```suggestion
"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
```

- Translation owners candidates, who are non-committers, MUST also meet the following criteria:
     - They are active long-term contributors to the Apache Airflow project at the time of request.
     - They have basic skills of working with GIT and GitHub, as well as modifying JSON translation files within their target language.
     - They have the support of an Apache Airflow committer who will act as a translation sponsor.
- Translation owners candidates MUST go through the approval process detailed in section 6.1.

### 5.3. Approval of new translations

To accept a new translation to the codebase, it MUST be approved through a review process discussed in section 6.2.

### 5.4. Resolution of translation conflicts

Translation conflicts MUST be resolved according to the procedures outlined in section 6.3.

### 5.5. Adding / rephrasing terms

- When new terms are added to the default language, all translation owners SHOULD create a follow-up PR to comply with the changes in their assigned language within a reasonable time.
- When rephrasing terms in the default language, all translation owners SHOULD do the same as above, **if needed**.

> - In busy times with many parallel UI changes it is acceptable to batch changes together, latest prior a release the differences SHOULD be cleared.

Reviewer: ```suggestion
- In busy times with many parallel UI changes it is acceptable to batch changes together. Differences SHOULD be cleared prior to a release at the latest.
```

    - Their own PRs should be approved and merged normally by another committer, preferably another code owner in their locale (if such exists).
- Otherwise, if the code owner is sponsored:
    - They should merge the translation-related PRs only after they are approved by the translation owner.
- Other committers may review and approve translation-related PRs in any aspects, but they SHOULD NOT merge them without the approval of the translation owner and the consent of the code owner.
- Before merging a translation-related PR, the translation should be checked for completeness using the provided tools (see section 8).

> [!WARNING]
> In languages with different word order than English, or in Right-To-Left (RTL) languages, it is important to validate that the changes are properly reflected in the UI.
> If they are not, please raise a GitHub issue or a PR for fixing it (separately from the translation PR).

### 5.8. Version release

- Requirements for release managers are defined in the [Release Management Policy](../../../../../../dev/README_RELEASE_AIRFLOW.md).

## 6. Procedures

### 6.1. Approval process of a translation owner

> - The designated code owner should post a thread in the dev. list to request for approval of the translation owner(s) for a supported translation:

Reviewer 6: Maybe we should also include sponsors to the thread and await approval from the sponsor in addition to binding votes rather than commenting in the Github?
Just and idea not a blocker

Author: In case of sponsors, they are the designated code owners and should run the approval process.
I"ll clarify it in the next ammendment.

  - Ownership for both roles MUST be approved according to the process discussed in section 6.1.
  - A single Apache Airflow committer MAY serve as both code owner and translation owner for the same locale.
  - If none of the translation owners are code owners - there MAY be a translation sponsor assigned as a code owner.
- When the above is not met, steps mentioned in section 6.4 SHOULD be taken by the appropriate roles.

> [!NOTE]
> It is welcomed and desired to have more than one translation owner to enable peer reviews and provide coverage during absences.

### 5.2. Adding new locales

To accept a new supported locale to the codebase, it MUST be approved through the process discussed in section 6.2.

### 5.3. Translation owners candidates

- Translation owners candidates MUST declare and demonstrate a sufficient level of proficiency in the target language for translation purposes, including technical terminology (as detailed in section 6.5).
- Translation owners candidates, who are non-committers, MUST also meet the following criteria:
     - They are active long-term contributors to the Apache Airflow project at the time of request.

>      - They have basic skills of working with GIT and GitHub, as well as modifying JSON translation files within their target language.

Reviewer 2: ```suggestion
     - They have basic skills of working with Git and GitHub, as well as modifying JSON translation files within their target language.
```

nit

- Before merging a translation-related PR, the translation SHOULD be checked for completeness using the provided tools (see section 8).

> [!WARNING]
> In languages with different word order than English, or in Right-To-Left (RTL) languages, it is important to validate that the changes are properly reflected in the UI.
> If they are not, please raise a GitHub issue or a PR for fixing it (separately from the translation PR).

### 5.8. Version release

- Release managers MUST follow the requirements for releasing changes in supported locales defined in the [Release Management Policy](../../../../../../dev/README_RELEASE_AIRFLOW.md).

## 6. Procedures

### 6.1. Approval of ownership candidates

- The designated code owner, should post a thread to the dev list, requesting the approval of:
  - Themselves as the code owner in the suggested locale. If they are a translation sponsor, they should indicate it as well.
  - Other code owner(s) in the suggested locale, if applicable.
  - The identity of the translation owner(s) in the suggested locale (which could be themselves as well)

> - When introducing a new locale, the thread should announce that (including a link to the PR)

Reviewer 2: ```suggestion
- When introducing a new locale, the thread should announce that on the dev list. (including a link to the PR)
```

I was a bit confused at first glance 🤔 not sure whether it's better this way

Author: it should be "the same thread"

- Within the thread, the code owner should demonstrate that the translation owner is suitable for the role, according to the requirements in section 5.3.
- Approval of a code owner who is also the translation owner can be done by a lazy consensus.
- Approval of any translation owner who is not a committer requires at least one binding vote of 1 PMC member, and no objections from other committers/PMC.

### 6.2. Approval of a new locale

The following steps outline the process for approving a new locale to be added to the supported locales:

- Creating a PR for adding the suggested locale to the codebase ([see example](https://github.com/apache/airflow/pull/51258/files)), which includes:

>     - The locale files (translated according to the guidelines) in the `airflow-core/src/airflow/ui/src/i18n/locales/<LOCALE_CODE>` directory, where `<LOCALE_CODE>` is the code of the language according to ISO 639-1 standard (e.g., `fr` for French). Languages with regional variants should be handled in separate directories, where the name is suffixed with `-<VARIANT>`, and `<VARIANT>` is the variant follows ISO 3166-1 or UN M.49 codes in lowercase (e.g., `zh-tw` for Taiwanese Chinese).

Reviewer 2: ```suggestion
    - The locale files (translated according to the guidelines) in the `airflow-core/src/airflow/ui/src/i18n/locales/<LOCALE_CODE>` directory, where `<LOCALE_CODE>` is the code of the language according to ISO 639-1 standard (e.g., `fr` for French). Languages with regional variants should be handled in separate directories, where the name is suffixed with `-<VARIANT>`, and `<VARIANT>` is the variant that follows ISO 3166-1 or UN M.49 codes in lowercase (e.g., `zh-tw` for Taiwanese Mandarin).
```

I re-read the policy and fixed my comment. I also noticed communities that use  IETF language tag  (BCP 47) with ISO 639-1
